terragrunt/modules/crates-io/compute-static/scripts/run_tests.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+# Start the Python HTTP server in the background, which will act as primary host for requests sent from Rust tests
+python3 scripts/test_http_server.py &
```

Comment:
the "real" way to write integration tests is actually another one, using [`viceroy-lib`](https://docs.rs/viceroy-lib/latest/viceroy_lib/), directly, running the WASM binary. It's a little more complex, needs a non-wasm subcrate for the binary, bun I once made it work for a project then you could just  use `mockito` to generate a test case specific mock-server. And it's a "real" integration test. 

My proposal for this PR would be: 
Keep it simple and just for unit-tests. It's already super helpful to be able to write normal unit-tests, without a real server. This small python server has only limited value for testing the many edge cases you want to test.

Reply:
I agree. I consider this implementation with the Python HTTP server hacky. I initially implemented it as I would've preferred to add coverage for the `send_request_to_s3` function, as it's the common case handled by this compute function.
I didn't think of using viceroy-lib directly, that seems definitely cleaner. I just saw it's also mentioned in [Viceroy's README](https://github.com/fastly/Viceroy?tab=readme-ov-file#usage-as-a-library) as the suggested way to run integration tests. Considering the need for a subcrate for this kind of test, I also think it might be worth considering whether this is actually needed, and if so having a separate PR - I'll let the maintainers evaluate whether they'd appreciate this or if it's overkill for now.